### PR TITLE
Add helical and annular shell trajectories

### PR DESCRIPTION
**Pull request by Daval-G**
This PR adds two 3D trajectories based on concentric shells:
- helical shells: a quite famous trajectory that consists of spirals covering concentric shells and that can be at least traced back to Irarrazabal, Pablo, and Dwight G. Nishimura. "Fast three dimensional magnetic resonance imaging." Magnetic Resonance in Medicine 33, no. 5 (1995): 656-662, but the implementation is based on the remarkably clear proposition by Shu, Yunhong, Stephen J. Riederer, and Matt A. Bernstein. "Three‐dimensional MRI with an undersampled spherical shells trajectory." Magnetic Resonance in Medicine 56, no. 3 (2006): 553-562. A future PR might propose a companion function to handle gradient constraint in the polar regions as suggested in the article.
- annular shell: an exclusive trajectory based on solutions to the sphere packing problem inspired from Gerlach, Henryk, and Heiko von der Mosel. "On sphere-filling ropes." The American Mathematical Monthly 118, no. 10 (2011): 863-876. The trajectory consists of shells composed of rings cut in half and rotated to create a continuity with other rings, and therefore obtain more homogeneous shots in term of length as compared to simply spherically stacking rings.

Both trajectory implementations might be changed in future PRs that will include new expansion methods/tools to "shellify" and spherically stack arbitrary 2D planes.

### CODE REVIEW

**Comment by Daval-G** [comment]
Pictures for representation:
- helical shell trajectory:

- annular shell trajectory:


Examples and documentation will come in later PRs, as there is no documentation on 3D trajectories currently and I plan an overhaul of the 2D-to-3D expansions.

File: src/mrinufft/trajectories/trajectory3D.py [hunk 1]
```diff
@@ -149,6 +150,180 @@ def initialize_3D_cones(
...
+    oversampling : int
+        Multiplicative factor for Ns without changing the curves, by default 1
+    tilt : str, float, optional
+        Angle between shots over a shell surface, by default "uniform"
+
+    Returns
+    -------
+    array_like
+        3D helical shell trajectory
+    """
```
Comment (paquiteau) [inline src/mrinufft/trajectories/trajectory3D.py:173]: Don't forget to add a Reference section, with citations. A Notes section with the history mentioned in the PR can also be added.

File: src/mrinufft/trajectories/trajectory3D.py [hunk 2]
```diff
@@ -149,6 +150,180 @@ def initialize_3D_cones(
...
+    shell_tilt : str, float, optional
+        Angle between consecutive shells, by default pi
+    ring_tilt :
+        Angle controlling approximately the shell halves rotation, by default pi / 2
+
+    Returns
+    -------
+    array_like
+        3D annular shell trajectory
+    """
```
Comment (paquiteau) [inline src/mrinufft/trajectories/trajectory3D.py:241]: Don't forget to add reference here also.

**Review by paquiteau — CHANGES_REQUESTED** [review]
Great Stuff again ! 

I guess it is still needed to add them to trajectory example script.

File: src/mrinufft/trajectories/trajectory3D.py (hunk 2 above)
Comment (Daval-G) [reply src/mrinufft/trajectories/trajectory3D.py:254]: Done

File: src/mrinufft/trajectories/trajectory3D.py (hunk 1 above)
Comment (Daval-G) [reply src/mrinufft/trajectories/trajectory3D.py:177]: Done

**Review by paquiteau — APPROVED** [review]

**Pull request merged**